### PR TITLE
build: jsonschema and protobuf versions updated

### PR DESCRIPTION
**Pull request by solarw**
## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request.

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](https://github.com/fetchai/agents-aea/blob/main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

DELETE INCLUSIVE THIS AND BELOW FOR STANDARD PR
------

## Release summary

Version number: [e.g. 1.0.1]

## Release details

Describe in short the main changes with the new release.

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](https://github.com/fetchai/agents-aea/blob/master/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side), from `develop`
- [ ] Lint and unit tests pass locally and in CI
- [ ] I have checked the fingerprint hashes are correct by running (`scripts/generate_ipfs_hashes.py`)
- [ ] I have regenerated the latest API docs
- [ ] I built the documentation and updated it with the latest changes
- [ ] I have added an item in `HISTORY.md` for this release
- [ ] I bumped the version number in the `aea/__version__.py` file.
- [ ] I bumped the version number in every Docker image of the repo and published it. Also, I built and published them with tag `latest`  
      (check the READMEs of [`aea-develop`](https://github.com/fetchai/agents-aea/blob/master/develop-image/README.md#publish) 
      and [`aea-user`](https://github.com/fetchai/agents-aea/blob/master/develop-image/user-image/README.md#publish))
- [ ] I have pushed the latest packages to the registry.
- [ ] I have uploaded the latest `aea` to PyPI.
- [ ] I have uploaded the latest plugins to PyPI.

## Further comments

Write here any other comment about the release, if any.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -37,10 +37,10 @@ aea = 'aea.cli:cli'
 [tool.poetry.dependencies]
 python = ">=3.8,<3.11"
 base58 = ">=1.0.3,<3.0.0"
-jsonschema = "^4.0.0"
+jsonschema = ">=3.2.0,<5"
 packaging = "^21.0"
 semver = ">=2.9.1,<3.0.0"
-protobuf = ">=3.19.0,<=3.19.4"
+protobuf = ">=3.19.4,<4"
```
Comment (5A11) [inline pyproject.toml:43]: Shouldn't we peg it to a specific version?

File: pyproject.toml [hunk 2]
```diff
@@ -37,10 +37,10 @@ aea = 'aea.cli:cli'
 [tool.poetry.dependencies]
 python = ">=3.8,<3.11"
 base58 = ">=1.0.3,<3.0.0"
-jsonschema = "^4.0.0"
+jsonschema = ">=3.2.0,<5"
```
Comment (5A11) [inline pyproject.toml:40]: Why from 3.2.0 to 5?

**Review by 5A11 — COMMENTED** [review]
Some CI failures.

File: pyproject.toml (hunk 1 above)
Comment (solarw) [reply pyproject.toml:43]: no, it's better to stay it like this

File: pyproject.toml (hunk 2 above)
Comment (solarw) [reply pyproject.toml:40]: it works ok and 3.2 required by external team

**Comment by mariacarmina** [comment]
> Some CI failures.

I encountered the same CI failures on my PR as well.

File: tests/data/reference_protocols/t_protocol_no_ct/__init__.py [hunk 3]
```diff
@@ -20,7 +20,7 @@
 """
 This module contains the support resources for the t_protocol_no_ct protocol.
 
-It was created with protocol buffer compiler version `libprotoc 3.19.4` and aea version `1.2.2`.
+It was created with protocol buffer compiler version `libprotoc 3.19.4` and aea version `1.2.3`.
```
Comment (5A11) [inline tests/data/reference_protocols/t_protocol_no_ct/__init__.py:23]: Why aren't all protocol regenerated with the new aea framework version 1.2.3?

**Review by 5A11 — APPROVED** [review]

**Review by mariacarmina — COMMENTED** [review]
Can we modify the `cosmpy` version to `">=0.6.1,<0.7.0"` as we have [here](https://github.com/fetchai/agents-aea/pull/2813/files#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711R68) and after kindly update the poetry.lock file as well?
Other than that, the conflicts should be solved as well and that would be ready to go.



sure, will set it explicitly

**Pull request merged**